### PR TITLE
Allow all available metric data types

### PR DESCRIPTION
**Pull request by tjoyal**
I was playing around the gem in order to allow the same metric types as the service.

screen shot 2017-08-31 at 8 11 42 pm

This is what it boiled down to:
```
dog = Dogapi::Client.new('token')
response = dog.emit_point('test.type.count', 1, host: "datacat", type: 'count')
puts response.inspect
=> ["202", {"status"=>"ok"}]
```

But the type didn't register properly in the backend:
screen shot 2017-08-31 at 8 16 09 pm

I can only presume the backend issue is why there is a hard limit for these 2 types in the ruby gem for now?
Could you advice on if/how I should pursue resolution of this issue?

Thank you

### CODE REVIEW

File: lib/dogapi/v1/metric.rb [hunk 1]
```diff
@@ -7,6 +7,12 @@ class V1 # for namespacing
     class MetricService < Dogapi::APIService
 
       API_VERSION = 'v1'
+      METRIC_TYPES = %w(
```
Comment (gzussa) [inline lib/dogapi/v1/metric.rb]: Some types are missing. See https://docs.datadoghq.com/developers/metrics/types/?tab=count

File: lib/dogapi/v1/metric.rb [hunk 2]
```diff
@@ -59,16 +65,16 @@ def switch_to_single()
 
       def make_metric_payload(metric, points, scope, options)
         begin
-          typ = options[:type] || 'gauge'
+          type = options[:type]
```
Comment (gzussa) [inline lib/dogapi/v1/metric.rb]: Let's add the interval param as well 😃 . See https://docs.datadoghq.com/api/?lang=bash#metrics

File: lib/dogapi/v1/metric.rb [hunk 3]
```diff
@@ -7,6 +7,12 @@ class V1 # for namespacing
     class MetricService < Dogapi::APIService
 
       API_VERSION = 'v1'
+      METRIC_TYPES = %w(
+        count
+        counter
```
Comment (jirikuncar) [inline lib/dogapi/v1/metric.rb]:
```suggestion
        rate
```

File: lib/dogapi/v1/metric.rb [hunk 4]
```diff
@@ -7,6 +7,12 @@ class V1 # for namespacing
...
+        gauge
+        rate
```
Comment (jirikuncar) [inline lib/dogapi/v1/metric.rb]:
```suggestion
        gauge
```

File: lib/dogapi/v1/metric.rb [hunk 5]
```diff
@@ -7,6 +7,12 @@ class V1 # for namespacing
...
+      )
```
Comment (jirikuncar) [inline lib/dogapi/v1/metric.rb]:
```suggestion
        histogram
        distribution
        set
        timers
      )
```

File: lib/dogapi/v1/metric.rb (hunk 3 above)
Comment (jirikuncar) [inline lib/dogapi/v1/metric.rb]:
```suggestion
```

**Comment by zippolyte** [comment]
Closing, this fix was merged with https://github.com/DataDog/dogapi-rb/pull/165
Thanks for contributing

**Comment by tjoyal** [comment]
Thank you for carrying it through the finish line. 💚

**Pull request closed without merging**